src/sorcha/sorcha.py

```diff
@@ -165,6 +166,10 @@ def runLSSTSimulation(args, configs, pplogger=None):
             observations = reader.read_block(block_size=configs["size_serial_chunk"])
         else:
             orbits_df = reader.read_aux_block(block_size=configs["size_serial_chunk"])
+            # determine if the pointing dataframe has been primed for creating ephemeris
+            if "visit_vector" not in filterpointing:
```

Comment:
Please let me know if there's a better way to guard this. Right now the code just checks for one of the columns that would have been added if the pre-computation step had already happened.

Reply:
I have generally used dictionaries for this kind of thing.  The way I have kept track of visits with multiple telescopes is to use a key that is a tuple of the observatory code and the date/time of the exposure.   That should always be unique and corresponds one-to-one with a set of the precalculations you are doing.  Not sure if that helps you, but maybe you can use that idea to help keep track of what has already been done.

Reply:
Why not just do this precomputation when to first create `filterpointing` (Around line 118)

You could still use the `configs["ephemerides_type"]` check from line 164.

Reply:
Ooohhh yeahhhh. Yes, that's a great point @jeremykubica thanks for the suggestion!